### DISCUSSION TITLE
Standalone minimalistic tsh

### DISCUSSION THREAD

**Discussion by rkouznetsov** [gravitational/teleport#17615] [Ideas]
Hi,
There are more and more services around which cannot be used without teleport. To use that from Ubuntu one needs to download and manually keep up-to-date .deb package, or to set up a 3-rd party repository. In any case one gets full package with server, GUI and all bells and whistles, that takes close to 500M of disk space when installed.  Both size and lack of standard-repo support might be critical in some applications.

Are there any plans to make a standalone minimalistic package e.g. teleport-tsh-client available via official Ubuntu repositories? 

It would be also great to have a version of tsh that can be set up and run without sudo/root access outside of containers.

Thank you!

**Comment by webvictim** [comment]
`tsh` doesn't need root access today - unless I'm misunderstanding what you're asking?

It's possible that we might eventually make a tsh client-only package but isn't a priority at the moment. I'd encourage you to build your own Docker image/package containing the binaries you need if size and download time is a concern for you.

**Reply by rkouznetsov** [reply]
Thank you! 

Root access is needed to set up automatic update of tsh together with regular system updates. 
Asking a guy with root privileges to do `sudo apt install tsh-client` (or whatever) is much easier than asking them to do all the magic described at https://goteleport.com/docs/installation/.  That is the main reason why availability from standard repos is desirable. At least there should be no legal issues with that, since the apache license is good enough for that.

After reading your reply I have spotted that tsh from the binary tarball (x86-64) is statically linked, so it can be just used "as is" without docker or anything alike. It is still about 20 times bigger than installed size of openssh-client form Ubuntu 20.04, but x5 times smaller than full teleport package, and seems to work. `gzexe` shaves yet another factor of 3 from the binary's size, which is already acceptable in terms of size. 

I hope non-updatable security software will not pose a major security threat until the standalone tsh package is available from Ubuntu repos :)

**Reply by webvictim** [reply]
`curl -s https://cdn.teleport.dev/teleport-v10.3.3-linux-amd64-bin.tar.gz | tar -xvzf - teleport/tsh` also works!

**Reply by rkouznetsov** [reply]
Thank you! That's pretty much what I did. :)

**Reply by claudenobs** [reply]
> `curl -s https://cdn.teleport.dev/teleport-v10.3.3-linux-amd64-bin.tar.gz | tar -xvzf - teleport/tsh`

this should really be the entirety of this webpage: https://goteleport.com/docs/connect-your-client/tsh/#installing-tsh

well, combined with the line that gets the version : 
```
v="$(curl -fsSL https://teleport.bedag.ch/webapi/find | jq -r '.server_version')"
curl -fsSL "https://cdn.teleport.dev/teleport-v$v-linux-amd64-bin.tar.gz" | tar -xvzf - teleport/tsh
```
